tests/test_grading.py

```diff
@@ -92,9 +92,9 @@ def test_course_submissions_optimized(self):
             print("Running with {}".format(db.engine.name))
             query = self.assignment.mysql_course_submissions_query()
             data = [d for d in query]
-            backups_from_query = [d[6] for d in data if d[6]]
+            backups_from_query = [d[7] for d in data if d[7]]
```

Comment:
I really really don't like the magic number 7 appearing here :( can course_submissions_query return a sequence of dicts somehow? It could be a generator so it doesn't destroy the pagination that sqlalchemy might do